backend/server.js

```diff
@@ -23,30 +24,137 @@ if (app.get('env') === 'development') {
+  .delete(checkDeleteAccountSchema, accounts.deleteAccount)
+function checkGetAccountsByIdSchema(req, res, next) {
+  const getAccountsByIdSchema = Joi.object().keys({
+    id: Joi.number().integer().required(),
+  })
+  validateRequest(req, next, getAccountsByIdSchema)
+}
+function checkUpdateAccountSchema(req, res, next) {
+  const updateAccountSchema = Joi.object().keys({
+    id: Joi.number().integer().required(),
+  })
+  validateRequest(req, next, updateAccountSchema)
+}
+function checkDeleteAccountSchema(req, res, next) {
+  const deleteAccountSchema = Joi.object().keys({
+    id: Joi.number().integer().required(),
+  })
+  validateRequest(req, next, deleteAccountSchema)
+}
 
-app.route('/username/:username').get(accounts.checkUsername)
+app.route('/username/:username').get(checkCheckUsernameSchema, accounts.checkUsername)
+function checkCheckUsernameSchema(req, res, next) {
+  const checkUsernameSchema = Joi.object().keys({
+    username: Joi.string().required(),
+  })
+  validateRequest(req, next, checkUsernameSchema)
+}
 
-app.route('/phoneNumber/:phone_number').get(accounts.checkPhoneNumber)
+app.route('/phoneNumber/:phone_number').get(checkCheckPhoneNumberSchema, accounts.checkPhoneNumber)
+function checkCheckPhoneNumberSchema(req, res, next) {
+  const phoneNumberSchema = Joi.object().keys({
+    phone_number: Joi.string().min(7).max(15),
+  })
+  validateRequest(req, next, phoneNumberSchema)
+}
 
 // friendships table
-app.route('/friendships').post(friends.createFriends)
+app.route('/friendships').post(checkCreateFriendsSchema, friends.createFriends)
+function checkCreateFriendsSchema(req, res, next) {
+  const createFriendsSchema = Joi.object()
+    .keys({
+      main: Joi.number().integer().required(),
+      friend: Joi.number().integer().required(),
+    })
+    .with('main', 'friend')
+  validateRequest(req, next, createFriendsSchema)
+}
 
-app.route('/friendships/friends/:user').get(friends.getFriends)
+app.route('/friendships/friends/:user').get(checkGetFriendsSchema, friends.getFriends)
+function checkGetFriendsSchema(req, res, next) {
+  const getFriendsSchema = Joi.object().keys({
+    m_id: Joi.number().integer().required(),
+  })
+  validateRequest(req, next, getFriendsSchema)
+}
 
 app
   .route('/friendships/friends/:user/:friend')
-  .delete(friends.deleteFriendship)
-  .put(friends.acceptRequest)
+  .delete(checkDeleteFriendshipSchema, friends.deleteFriendship)
+  .put(checkAcceptRequestSchema, friends.acceptRequest)
+function checkDeleteFriendshipSchema(req, res, next) {
+  const deleteFriendshipSchema = Joi.object()
+    .keys({
+      main: Joi.number().integer().required(),
+      friend: Joi.number().integer().required(),
+    })
+    .with('main', 'friend')
+  validateRequest(req, next, deleteFriendshipSchema)
+}
+function checkAcceptRequestSchema(req, res, next) {
+  const acceptRequestSchema = Joi.object()
+    .keys({
+      main: Joi.number().integer().required(),
```

Comment:
i believe this put request only puts parameters in the end points not the request body